userspace/libscap/engine/gvisor/gvisor.h

```diff
@@ -74,4 +79,33 @@ typedef struct parse_result parse_result;
 parse_result parse_gvisor_proto(scap_const_sized_buffer gvisor_buf, scap_sized_buffer scap_buf);
 
 } // namespace parsers
+
+
+class engine {
+public:
+    engine(char *lasterr);
+    ~engine();
+    int32_t init(std::string socket_path);
+    int32_t close();
+
+    int32_t start_capture();
+    int32_t stop_capture();
+
+    int32_t next(scap_evt **pevent, uint16_t *pcpuid);
+    
+private:
+    int32_t process_message_from_fd(int fd);
+    void free_sandbox_buffers();
+
+    char *m_lasterr;
+    int m_listenfd;
+    int m_epollfd;
+    std::string m_socket_path;
+    std::thread m_accept_thread;
+    std::deque<scap_evt *> m_event_queue{};
+
+    // buffers in which to store events, one per each active sandbox, indexed by fd
+    std::map<int, scap_sized_buffer> m_sandbox_buffers;
```

Comment:
1. `std::unordered_map`? should be faster
2. can we wrap the `scap_sized_buffer` in a C++ type with a destructor? Looks like we're doing manual cleanup in multiple places

Reply:
1. and 2. done, makes sense! Although I don't think it'd make huge difference because I'm not expecting a very large number of sandboxes to be active at any one time.